### PR TITLE
[release/8.0-rc1] Revert changes to HasNativeCodeReJitAware

### PR DESCRIPTION
**Pull request by mikelle-rogers**
Changes caused test failures.

This commit reverts changes to the commit eacb32e400b3f9d3522bed193bd534d02e8b170a
We found that the previously mentioned commit, eacb32e, causes failures in the EnC tests. In addition, the commit eacb32e is not a high stakes commit, so it can easily be fixed and added to RC2. 

## Customer Impact
The commit eacb32e400b3f9d3522bed193bd534d02e8b170a causes failures in the EnC tests. 

## Testing
The commit eacb32e400b3f9d3522bed193bd534d02e8b170a causes failures in the EnC tests. 
## Risk
The risk of not merging the reverted commit is that EnC will not work reliably.

### CODE REVIEW

**Review by tommcdon — APPROVED** [review]

**Comment by mikelle-rogers** [comment]
I just verified that reverting this commit results in the tests passing.

**Review by hoyosjs — APPROVED** [review]

**Review by jeffschwMSFT — APPROVED** [review]
approved.  we will take for consideration in RC1

**Comment by carlossanlop** [comment]
CI failure is known: https://github.com/dotnet/runtime/issues/90593

I'm waiting for Tactics approval to merge, I saw your email.

**Comment by carlossanlop** [comment]
Approved by Tactics via email.

**Pull request merged**
